### PR TITLE
feature delete default etcd data in deinit

### PR DESCRIPTION
**Pull request by Momeaking**
Signed-off-by: Momeaking <1026415175@qq.com>

**What type of PR is this?**

kind feature

**What this PR does / why we need it**:

Delete ETCD data during deinit

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
karmadactl deinit --de-etcd
```

### CODE REVIEW

File: pkg/karmadactl/deinit.go [hunk 1]
```diff
@@ -294,7 +319,16 @@ func (o *CommandDeInitOption) Run() error {
...
 
-	fmt.Println("remove Karmada from Kubernetes successfully.\n" +
-		"\ndeinit will not delete etcd data, if the etcd data is persistent, please delete it yourself.")
+	if o.DeDefaultEtcd {
+		err := cleanDir(etcdDefaultPath)
+		if err != nil {
+			return err
+		}
+		fmt.Println("remove Karmada from Kubernetes successfully.\n" +
+			"\n delete etcd data  in the default path,default path is : /var/lib/karmada-etcd .")
```
Comment (Poor12) [inline pkg/karmadactl/deinit.go:328]:
```suggestion
			"\ndelete etcd data in the default path, default path is: /var/lib/karmada-etcd.")
```

File: pkg/karmadactl/deinit.go [hunk 2]
```diff
@@ -294,7 +319,16 @@ func (o *CommandDeInitOption) Run() error {
...
+	} else {
+		fmt.Println("remove Karmada from Kubernetes successfully.\n" +
+			"\ndeinit will not delete etcd data, if the etcd data is persistent, please delete it yourself.")
```
Comment (Poor12) [inline pkg/karmadactl/deinit.go:331]:
```suggestion
			"\ndeinit will not delete etcd data, if the etcd data is persistent, please delete it by yourself.")
```

**Comment by prodanlabs** [comment]
There is a problem, if the host cluster has multiple nodes, and etcd is not running the current node, then this command is invalid.

**Comment by prodanlabs** [comment]
By the way, if the persistence of `ectd` uses `pvc`, because after deleting the `namespace`, `pvc` will be deleted in cascade, so it is wrong to say that `deinit` will not delete `etcd` data.

**Comment by Momeaking** [comment]
> By the way, if the persistence of `ectd` uses `pvc`, because after deleting the `namespace`, `pvc` will be deleted in cascade, so it is wrong to say that `deinit` will not delete `etcd` data.

First, the --t parameters are optional, and the user can choose whether to configure them at uninstall time based on how etCD data is persisted
Second,if the persistence of `ectd` uses `pvc`, you can leave this parameter out,delete it by yourself

**Comment by prodanlabs** [comment]
> First, the --t parameters are optional, and the user can choose whether to configure them at uninstall time based on how etCD data is persisted Second,if the persistence of `ectd` uses `pvc`, you can leave this parameter out,delete it by yourself

agree.

Another small question, what if `etcd` is not on the current server.

**Comment by Momeaking** [comment]
> delete it by yourself



> > First, the --t parameters are optional, and the user can choose whether to configure them at uninstall time based on how etCD data is persisted Second,if the persistence of `ectd` uses `pvc`, you can leave this parameter out,delete it by yourself
> 
> agree.
> 
> Another small question, what if `etcd` is not on the current server.

If this is the case, you can only delete it by yourself. In addition,This parameter is designed for installation using karmada init, similar to kubeadm init and kubeadm reset

**Comment by prodanlabs** [comment]
> If this is the case, you can only delete it by yourself. In addition,This parameter is designed for installation using karmada init, similar to kubeadm init and kubeadm reset

Or output some logs, prompting the user to delete it manually.

**Comment by RainbowMango** [comment]
I'm going to close this PR as it has been inactive for a long time. If you think this still needed, please feel free to re-open it.

In my opinion, there are plenty of ways people use ETCD, and the default persistence patch is unlikely to be used in production. 
So, it's complicated to consider the situation when and how to remove the ETCD data.
I want to hear more feedback from real-world use cases, and then consider how to enhance it.

Anyway, thanks for bring this. @Momeaking 

/close

**Pull request closed without merging**